### DISCUSSION TITLE
How to group results by year?

### DISCUSSION THREAD

**Discussion by holmesadrianh** [directus/directus#2777] [Feature Requests]
I am not sure where to begin with this requirement. I need to query all items from a collections, ordered by date, and then have them grouped by year. I am not sure if this is something that can be done with Directus or if I need to use PHP to do this. Is there any suggestion on the best way forward with this approach?

Any suggestions is much appreciated.

**Comment by benhaynes** [comment]
Really interesting... I think a `group` or `groupby` parameter in the API would be really useful. I can also already imagine how that would manifest in the App. I don't know if this would make it into 9.0, but perhaps 9.x if the Feature Request gets enough attention. 👍

**Comment by rijkvanzanten** [comment]
What would the ideal JSON output be of such a parameter?

**Reply by benhaynes** [reply]
Maybe the same as now, but with an additional layer of group objects in the response? Or are you talking about the param itself?

**Reply by rijkvanzanten** [reply]
Mostly how the JSON structure differs once you provide the field to group by

**Comment by holmesadrianh** [comment]
So what I wanted was to be able to separate the years. Each year had a collection of items. I found a solution with PHP, but not sure if its efficient or if there is a way to do it with Directus. I think it would be ideal if the original data was already structured.

Here is a my current solution:

```
$data = json_decode($content);
$years = [];
foreach ($data->data as $i => $item) {
    $date = explode('-', $item->publication_date);
    if (!isset($years[$date[0]])) {
        $years[$date[0]] = [];
    }
    array_push($years[$date[0]], $item);
}
```
Then the $years array groups them accordingly. This seems to be all I needed for this scenario. However, again, I imagine being able to avoid this kind of reorganising of the structure would be ideal.

**Reply by holmesadrianh** [reply]
I get this scenario is a bit unfair to expect. Here I wanted to have a specific number for the array key. But another approach that comes to mind is if the content editor could select a separate year function or lets say a dropdown to group it. And then sorting based on that dropdown. I do not mind if the content editor needs to select the group, but to be able to return the structure already grouped would be cool.

However, I absolutely love Directus for what it is. I have no complaints.

**Reply by rijkvanzanten** [reply]
Luckily this is easy enough to do in user land ofter you fetch the items, but I'm down to discuss the possibility for a dedicated `group` parameter that can do a similar thing :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
